packages/dds/tree/src/shared-tree/schematizedTree.ts

```diff
@@ -0,0 +1,124 @@
+): ISharedTreeBranch {
+	// Check for empty.
+	// This case could be better handled by making a schematized factory (so it runs only in the "create" case),
+	// but handling it here works well enough, and can be considered a special case of an out of schema handler.
+	// Note that if this becomes a more proper out of schema handler, it should be made lazy.
+	{
+		if (tree.context.root.length === 0 && schemaDataIsEmpty(tree.storedSchema)) {
+			tree.storedSchema.update(config.schema);
+			tree.root = config.initialTree;
+		}
+	}
+
+	// TODO: support adapters and include them here.
+	const viewSchema = new ViewSchema(defaultSchemaPolicy, {}, config.schema);
+	{
+		const compatibility = viewSchema.checkCompatibility(tree.storedSchema);
+		switch (config.allowedSchemaModifications) {
+			case AllowedUpdateType.None: {
+				if (compatibility.read !== Compatibility.Compatible) {
+					fail(
+						"Existing stored schema permits trees which are incompatible with the view schema",
+					);
+				}
+
+				if (compatibility.write !== Compatibility.Compatible) {
+					// TODO: support readonly mode in this case.
+					fail("View schema permits trees which are incompatible with the stored schema");
+				}
+
+				break;
+			}
+			case AllowedUpdateType.SchemaCompatible: {
+				if (compatibility.read !== Compatibility.Compatible) {
+					fail(
+						"Existing stored schema permits trees which are incompatible with the view schema, so schema can not be updated",
+					);
+				}
+				if (compatibility.write !== Compatibility.Compatible) {
+					tree.storedSchema.update(config.schema);
+				}
+
+				break;
+			}
+			default: {
+				unreachableCase(config.allowedSchemaModifications);
+			}
+		}
+	}
+
+	// TODO: errors thrown by this will usually be in response to remote edits, and thus may not surface to the app.
+	// Two fixes should be done related to this:
+	// 1. Ensure errors in response to edits like this crash app and report telemetry.
+	// 2. Replace these (and the above) exception based errors with
+	// out of schema handlers which update the schematized view of the tree instead of throwing.
+	tree.storedSchema.registerDependent(
+		new SimpleObservingDependent(() => {
+			const compatibility = viewSchema.checkCompatibility(tree.storedSchema);
+			if (compatibility.read !== Compatibility.Compatible) {
+				fail(
+					"Stored schema changed to one that permits data incompatible with the view schema",
+				);
+			}
+
+			if (compatibility.write !== Compatibility.Compatible) {
+				// TODO: support readonly mode in this case.
+				fail(
+					"Stored schema changed to one that does not support all data allowed by view schema",
+				);
+			}
+		}),
+	);
+
+	return tree;
+}
+
+/**
+ * Options used to schematize a `SharedTree`
+ * @alpha
+ */
+export interface SchematizeConfiguration {
+	readonly schema: ViewSchemaCollection;
+	readonly allowedSchemaModifications: AllowedUpdateType;
+	readonly initialTree: ContextuallyTypedFieldData;
```

Comment:
Consider putting a doc on this one explaining it is only used in the case when the schematized tree has no data and no stored schema. In the case where a tree has no data, but _does_ have a stored schema which is compatible with the new view schema, a user might still expect the view to have its content initialized to this.

Reply:
done.